helper/src/main/java/me/lucko/helper/serialize/CircularRegion.java

```diff
@@ -0,0 +1,156 @@
+        double radius = object.get("radius").getAsDouble();
+
+        return of(center, radius);
+    }
+
+    public static CircularRegion of(BlockPosition center, double radius) {
+        Objects.requireNonNull(center, "center");
+        if (radius <= 0) throw new IllegalArgumentException("radius cannot be negative");
+        return new CircularRegion(center, radius);
+    }
+
+    private final BlockPosition center;
+    private final double radius;
+    private final double diameter;
+
+    private CircularRegion(BlockPosition center, double radius) {
+        this.center = center;
+        this.radius = radius;
+        this.diameter = radius * 2;
+    }
+
+    /**
+     * Determines if the specified {@link BlockPosition} is within the region
+     * @param pos target position
+     * @return true if the position is in the region
+     */
+    public boolean inRegion(BlockPosition pos) {
+        Objects.requireNonNull(pos, "pos");
+        return pos.toVector().distanceSquared(this.center.toVector()) < this.radius * this.radius;
+    }
+
+    /**
+     * Determines if the specified {@link Block} is within the region
+     * @param block target block
+     * @return true if the block is in the region
+     */
+    public boolean inRegion(Block block) {
+        Objects.requireNonNull(block, "block");
+        return block.getLocation().distanceSquared(this.center.toLocation()) < this.radius * radius;
+    }
+
+    /**
+     * The center of the region as a {@link BlockPosition}
+     * @return the center
+     */
+    public BlockPosition getCenter() {
+        return this.center;
+    }
+
+    /**
+     * The radius of the region
+     * @return the radius
+     */
+    public double getRadius() {
+        return this.radius;
+    }
+
+    /**
+     * The diameter of the region
+     * @return the diameter
+     */
+    public double getDiameter() {
+        return this.diameter;
+    }
+
+    /**
+     * The circumference of the region
+     * @return the circumference
+     */
+    public double getCircumference() {
+        return 2 * Math.PI * this.radius;
+    }
+
+    /**
+     * Get the circumference {@link BlockPosition} of the region
+     * @return the {@link BlockPosition}s
+     */
+    @Nonnull
+    public Set<BlockPosition> getOuterPositions() {
+        Set<BlockPosition> positions = new HashSet<>((int) getCircumference());
+        for (int degree = 0; degree < 360; degree++) {
+            float radian = Maths.toRadians(degree);
+
+            double x = Maths.cos(radian) * this.radius;
+            double z = Maths.sin(radian) * this.radius;
+
+            positions.add(this.center.add((int) x, 0, (int) z));
+        }
+        return Collections.unmodifiableSet(positions);
+    }
+
+    /**
+     * Resize the current region with a new radius. The enter remains the same
+     * @param radius the radius
+     * @return circular region with new radius
+     */
+    public CircularRegion resize(double radius) {
```

Comment:
Not sure this method is needed - better imo to just explicitly call the static constructor to resize an existing region.